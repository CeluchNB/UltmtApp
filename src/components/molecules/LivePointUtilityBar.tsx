import { IconButton } from 'react-native-paper'
import React from 'react'
import { useTheme } from '../../hooks'
import { ActivityIndicator, StyleSheet, Text, View } from 'react-native'

interface LivePointUtilityBarProps {
    undoDisabled: boolean
    loading: boolean
    error?: string
    onUndo: () => void
    onEdit: () => void
}

const LivePointUtilityBar: React.FC<LivePointUtilityBarProps> = ({
    undoDisabled,
    loading,
    error,
    onUndo,
    onEdit,
}) => {
    const {
        theme: { colors },
    } = useTheme()

    const styles = StyleSheet.create({
        headerContainer: {
            flexDirection: 'row',
            alignSelf: 'flex-end',
        },
        error: {
            color: colors.error,
            width: '75%',
        },
    })

    return (
        <View style={styles.headerContainer}>
            <Text style={styles.error}>{error}</Text>
            {!error && loading && (
                <ActivityIndicator color={colors.textPrimary} size="small" />
            )}
            <IconButton
                icon="undo"
                iconColor={colors.textPrimary}
                onPress={onUndo}
                disabled={undoDisabled}
                testID="undo-button"
            />
            <IconButton
                icon="pencil"
                iconColor={colors.textPrimary}
                onPress={onEdit}
                testID="edit-button"
            />
        </View>
    )
}

export default LivePointUtilityBar
