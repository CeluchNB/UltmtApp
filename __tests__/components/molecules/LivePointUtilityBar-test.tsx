import LivePointUtilityBar from '../../../src/components/molecules/LivePointUtilityBar'
import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'

describe('LivePointStatus', () => {
    it('matches base snapshot', () => {
        const onUndo = jest.fn()
        const snapshot = render(
            <LivePointUtilityBar
                undoDisabled={false}
                loading={false}
                onUndo={onUndo}
                onEdit={() => {}}
            />,
        )

        const undoBtn = snapshot.getByTestId('undo-button')
        fireEvent.press(undoBtn)
        expect(onUndo).toHaveBeenCalled()
        expect(snapshot.toJSON()).toMatchSnapshot()
    })

    it('matches snapshot while loading', () => {
        const onUndo = jest.fn()
        const snapshot = render(
            <LivePointUtilityBar
                undoDisabled={true}
                loading={true}
                onUndo={jest.fn()}
                onEdit={() => {}}
            />,
        )

        const undoBtn = snapshot.getByTestId('undo-button')
        fireEvent.press(undoBtn)
        expect(onUndo).not.toHaveBeenCalled()
        expect(snapshot.toJSON()).toMatchSnapshot()
    })

    it('presses edit button', () => {
        const onEdit = jest.fn()
        const snapshot = render(
            <LivePointUtilityBar
                undoDisabled={true}
                loading={false}
                onUndo={jest.fn()}
                onEdit={onEdit}
            />,
        )

        const editBtn = snapshot.getByTestId('edit-button')
        fireEvent.press(editBtn)
        expect(onEdit).toHaveBeenCalled()
    })
})
