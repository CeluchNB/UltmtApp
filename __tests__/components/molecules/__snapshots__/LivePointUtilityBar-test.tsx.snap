// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LivePointStatus matches base snapshot 1`] = `
<View
  style={
    {
      "alignSelf": "flex-end",
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      {
        "color": "#FFB4AB",
        "width": "75%",
      }
    }
  />
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "transparent",
        "borderRadius": 20,
        "height": 40,
        "margin": 6,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "width": 40,
      }
    }
    testID="undo-button-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 20,
          "borderWidth": 0,
          "elevation": 0,
          "flex": 1,
          "overflow": "hidden",
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="undo-button-container"
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityTraits="button"
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        centered={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "alignItems": "center",
                "flexGrow": 1,
                "justifyContent": "center",
              },
              {
                "borderRadius": 20,
              },
            ],
          ]
        }
        testID="undo-button"
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            [
              {
                "color": "#ffd600",
                "fontSize": 24,
              },
              [
                {
                  "lineHeight": 24,
                  "transform": [
                    {
                      "scaleX": 1,
                    },
                  ],
                },
                {
                  "backgroundColor": "transparent",
                },
              ],
              {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          󰕌
        </Text>
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "transparent",
        "borderRadius": 20,
        "height": 40,
        "margin": 6,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "width": 40,
      }
    }
    testID="edit-button-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 20,
          "borderWidth": 0,
          "elevation": 0,
          "flex": 1,
          "overflow": "hidden",
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="edit-button-container"
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityTraits="button"
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        centered={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "alignItems": "center",
                "flexGrow": 1,
                "justifyContent": "center",
              },
              {
                "borderRadius": 20,
              },
            ],
          ]
        }
        testID="edit-button"
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            [
              {
                "color": "#ffd600",
                "fontSize": 24,
              },
              [
                {
                  "lineHeight": 24,
                  "transform": [
                    {
                      "scaleX": 1,
                    },
                  ],
                },
                {
                  "backgroundColor": "transparent",
                },
              ],
              {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          󰏫
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`LivePointStatus matches snapshot while loading 1`] = `
<View
  style={
    {
      "alignSelf": "flex-end",
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      {
        "color": "#FFB4AB",
        "width": "75%",
      }
    }
  />
  <ActivityIndicator
    color="#ffd600"
    size="small"
  />
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "transparent",
        "borderRadius": 20,
        "height": 40,
        "margin": 6,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "width": 40,
      }
    }
    testID="undo-button-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderColor": "rgba(28, 27, 31, 0.12)",
          "borderRadius": 20,
          "borderWidth": 0,
          "elevation": 0,
          "flex": 1,
          "overflow": "hidden",
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="undo-button-container"
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityTraits={
          [
            "button",
            "disabled",
          ]
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        centered={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "alignItems": "center",
                "flexGrow": 1,
                "justifyContent": "center",
              },
              {
                "borderRadius": 20,
              },
            ],
          ]
        }
        testID="undo-button"
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            [
              {
                "color": "rgba(28, 27, 31, 0.38)",
                "fontSize": 24,
              },
              [
                {
                  "lineHeight": 24,
                  "transform": [
                    {
                      "scaleX": 1,
                    },
                  ],
                },
                {
                  "backgroundColor": "transparent",
                },
              ],
              {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          󰕌
        </Text>
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "transparent",
        "borderRadius": 20,
        "height": 40,
        "margin": 6,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "width": 40,
      }
    }
    testID="edit-button-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 20,
          "borderWidth": 0,
          "elevation": 0,
          "flex": 1,
          "overflow": "hidden",
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="edit-button-container"
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityTraits="button"
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        centered={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "alignItems": "center",
                "flexGrow": 1,
                "justifyContent": "center",
              },
              {
                "borderRadius": 20,
              },
            ],
          ]
        }
        testID="edit-button"
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            [
              {
                "color": "#ffd600",
                "fontSize": 24,
              },
              [
                {
                  "lineHeight": 24,
                  "transform": [
                    {
                      "scaleX": 1,
                    },
                  ],
                },
                {
                  "backgroundColor": "transparent",
                },
              ],
              {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          󰏫
        </Text>
      </View>
    </View>
  </View>
</View>
`;
